src/main/java/com/redhat/devtools/lsp4ij/ui/LSP4IJUiUtils.java

```diff
@@ -0,0 +1,116 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at https://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+
+package com.redhat.devtools.lsp4ij.ui;
+
+import com.intellij.codeInsight.hint.HintManager;
+import com.intellij.codeInsight.hint.HintManagerImpl;
+import com.intellij.codeInsight.hint.HintUtil;
+import com.intellij.openapi.application.ApplicationManager;
+import com.intellij.openapi.editor.Editor;
+import com.intellij.openapi.wm.impl.status.StatusBarUtil;
+import com.intellij.psi.PsiFile;
+import com.intellij.ui.LightweightHint;
+import com.redhat.devtools.lsp4ij.LSPIJUtils;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.swing.*;
+import java.util.function.Function;
+
+/**
+ * Common UI utilities.
+ */
+public final class LSP4IJUiUtils {
```

Comment:
Added a new pure utility class for UI-related utility methods. Right now it's just the `show*Hint()` family of methods, of which only one, `showErrorHint()`, is used directly. I can remove the unused ones if you'd like, but they're very simple and could well be useful for showing other types of editor hints. Let me know your thoughts.